chore: add CONTRIBUTING.md

tkurki opened the pull request:
Add instructions relating to PR flow and commit
messages. These have been unformalised so far. It
is a bit nasty to ask for changes in PRs when these
have been nowhere to find. Partially copied from
Angular DEVELOPERS.md.

webmasterkai commented:
This looks good to me. I might suggest we move the file to `/docs` so that it automatically shows up on issue creation for new users to review. https://help.github.com/articles/setting-guidelines-for-repository-contributors/

I like the **Pull Request and Commit messages** section https://github.com/SignalK/specification/blob/master/CONTRIBUTING.md and how it lists the <type> enums.

Might also be worth having an example `git commit` command to go along withe the others you have there.

tkurki commented:
- [x] moved to docs
- [x] added inline commit message guidelines in addition to the link

If you can't find your way to `git commit` then this doc has no hope - I want to include the stuff that even a bit more experienced people have trouble with.
